src/language-js/embed/graphql.js

```diff
@@ -108,6 +108,9 @@ function printGraphqlComments(lines) {
 function isGraphQL({ node, parent }) {
   return (
     hasLanguageComment(node, "GraphQL") ||
+    (parent &&
+      isAsConstExpression(parent) &&
+      hasLanguageComment(parent, "GraphQL")) ||
```

Comment:
I think we can move parent check logic into  `hasLanguageComment` for easier use.